### DISCUSSION TITLE
SumatraPDF pre-release build version 14547 very slow

### DISCUSSION THREAD

**Discussion by ltsnow** [sumatrapdfreader/sumatrapdf#2541] [General]
This version of SumatraPDF is very slow in opening PDF files, expecially larger ones.  The previous pre-release (14531) was fast as always.

**Comment by kjk** [comment]
I didn't make changes that should affect performance.

I've tried 14547 with a couple of large PDFs and they render instantly for me, as before.

You would have to provide a sample PDF that shows the problem and quantify the difference in speed (i.e. what is "slow"? seconds? minutes?)

**Reply by ltsnow** [reply]
[Camp Susquehanna.pdf](https://github.com/sumatrapdfreader/sumatrapdf/files/8643603/Camp.Susquehanna.pdf)

It happens with all my PDF files, but here is good example of a 4mb file that now takes about 3 seconds to load with 14547, but opens instantly with previous builds.

**Reply by kjk** [reply]
This file loads in 11 millisecond for me and pages render in under 100 ms.

You can see exact times if you upgrade to 14551 or later and then:
* open the file
* use command palette (`Ctrl + K`) and pick `Show log` option

In the log you should see sth like:
```
LoadDocument: 10.30 ms, 24 pages for 'C:\Users\kjk\Downloads\Camp.Susquehanna.pdf'
Slow rendering: 198.08 ms, page: 7 in 'C:\Users\kjk\Downloads\Camp.Susquehanna.pdf'
```

I only log page rendering time if longer than 100 ms (so you might not see those).

Please post the times you're seeing.

One explanation for slower loading: the file is on a network drive (including OneDrive, Dropbox etc.).

In such cases the limiting factor might be the speed of networking. You can see if there's a difference if you copy to a fixed C;\ drive and open from there.

14551 should be available within an hour for update check (and is already available in https://www.sumatrapdfreader.org/prerelease)

**Reply by ltsnow** [reply]
Thanks.  I'm getting about 4.00ms which is good, but I'm also getting crashes on every file I open.  The crash info folder has 75mgs of files and the log is referencing all sorts of dlls.  So even though the load time says 4.00ms it is really quite a bit longer because of the crashes. BTW, everything is fine if you load the files after opening SumatraPDF.  It only happens when you click on the PDF file without the program running.

NEW EDIT: I have found out what crashes SumatraPDF.  If I have default zoom set to Fit Width, Fit Content, or Fit Page it crashes.  If I use any other default zoom level it is fine.  I like using Fit Width and this is causing crashes.  This is with default layout set to continuous.

**Reply by GitHubRulesOK** [reply]
Hmm I have not seen crash due to width so lets see if I try that setting, however this brand new unit says it wants update video driver but warns that its not approved by OEM so I don't update it!

@kjk OK it crashed not with previous version on setting fit width  as default but after update to newer with it set
[sumatrapdfcrash.zip](https://github.com/sumatrapdfreader/sumatrapdf/files/8644181/sumatrapdfcrash.zip)

**Reply by kjk** [reply]
thanks, the crash should be fixed in 14553 or later

**Reply by ltsnow** [reply]
Yes, it is fixed in 14553.  However, FYI you still have to select Use Tabs in advanced or it will still crash.  No big deal for me since I don't ever use tabs.

**Reply by kjk** [reply]
@ltsnow could you provide crash report? I think I've fixed the crash in crash report by @GitHubRulesOK above but you might be seeing a different crash which I cannot reproduce locally.

**Reply by ltsnow** [reply]
[sumatra-log.txt](https://github.com/sumatrapdfreader/sumatrapdf/files/8644431/sumatra-log.txt)

Yes, here is the crash log when Use Tabs is NOT selected.

**Reply by kjk** [reply]
thanks, I can reproduce it now locally. I've opened https://github.com/sumatrapdfreader/sumatrapdf/issues/2543 to track this
